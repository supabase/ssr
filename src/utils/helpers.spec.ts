import { describe, it, expect } from "vitest";

import {
  serialize,
  parse,
  // intentionally importing ^ to make sure they're exported, remove in v1.0.0
  parseCookieHeader,
  serializeCookieHeader,
} from "./helpers";

describe("helpers", () => {
  describe("parseCookieHeader", () => {
    it("should parse a Cookie header", () => {
      expect(parseCookieHeader("")).toMatchSnapshot();
      expect(
        parseCookieHeader(`a=b;c=${encodeURIComponent(" hello ")};e=f`),
      ).toMatchSnapshot();
    });
  });

  describe("serializeCookieHeader", () => {
    it("should serialize a cookie to a Set-Cookie header", () => {
      expect(
        serializeCookieHeader("a", "", {
          path: "/",
          maxAge: 123,
          httpOnly: true,
          secure: true,
        }),
      ).toMatchSnapshot();
      expect(
        serializeCookieHeader("b", " weird value", { maxAge: 345 }),
      ).toMatchSnapshot();
    });
  });
});
