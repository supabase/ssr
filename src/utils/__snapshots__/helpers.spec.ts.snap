// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`helpers > parseCookieHeader > should parse a Cookie header 1`] = `[]`;

exports[`helpers > parseCookieHeader > should parse a Cookie header 2`] = `
[
  {
    "name": "a",
    "value": "b",
  },
  {
    "name": "c",
    "value": " hello ",
  },
  {
    "name": "e",
    "value": "f",
  },
]
`;

exports[`helpers > serializeCookieHeader > should serialize a cookie to a Set-Cookie header 1`] = `"a=; Max-Age=123; Path=/; HttpOnly; Secure"`;

exports[`helpers > serializeCookieHeader > should serialize a cookie to a Set-Cookie header 2`] = `"b=%20weird%20value; Max-Age=345"`;
